Formula/m4.rb

```diff
@@ -32,6 +32,22 @@ class M4 < Formula
     end
   end
 
+  on_linux do
+    if OS::Linux::Glibc.system_version >= "2.28" ||
+       (Formula["glibc"].any_version_installed? && Formula["glibc"].version >= "2.28")
```

Comment:
This line appears to throw the error `No available formula with the name "glibc".` in CI.

`glibc` is specific, I guess, to Linux, so I expected `on_linux` to avoid triggering errors related to `glibc` as part of any Mac tests.

Should I just exclude the `Formula["glibc"]` checks?

The original intend was just to resolve an issue when relying on the system `glibc` version.

Reply:
I think so, yes. The Linux m4 formula doesn't depend on the globe formula (unless I misread it), so I don't think it's going to matter what version the installed brew glibc is.

Reply:
Weird, that shouldn't be happening. @hutson, try rebasing against master and force-push to trigger another CI round.

Reply:
I just removed the `Formula` reference **and** rebased. Whoops. :sweat: 

Let me revert that change and try again.

Reply:
Added back the `Formula` reference. Waiting to see if `tap_syntax` check passes or fails.

Reply:
Failed with the same error. Going back to removing the `Formula` reference.

Reply:
After removing `Formula` reference builds now all pass successfully.

Reply:
Looks like the tap_syntax test doesn't respect `on_macos` or `on_linux` blocks when checking formulae?

Not sure that's intentional; but perhaps I've misunderstood what's going on. Pinging @iMichka.

Reply:
> Looks like the tap_syntax test doesn't respect on_macos or on_linux blocks when checking formulae?

That's the thing. The only way is to add the glibc formula to this repo (which we plan to do). Or avoid calling glibc in a homebrew-core formula for the time being. I think it is a legit warning: even if the code executes only on linux, the formula should be in this repo.

Reply:
I ported this PR over to `linuxbrew-core` since I'm technically on Linux and I can reference `glibc` since the formula exists in that repository.

I guess I see three options here:
- Merge the PR "as is" to minimize the difference between `homebrew-core` and `linuxbrew-core`.
- Include a port of the `glibc` formula and the formula check.
- Close the PR and just expect `linuxbrew-core` will be merged back in at some point.

Thoughts?

Reply:
I added some comments on you linuxbrew-core PR that need some answers before we decide what we do.

Reply:
I'd suggest either removing the conditional `if OS::Linux::Glibc.system_version >= "2.28"` or waiting for an upstream release of `m4` that fixes this issue.

Reply:
> or waiting for an upstream release of m4 that fixes this issue.

As noted on the Linux PR:

The `gnulib` submodule for `m4` was updated on July 5th, 2020 - https://git.savannah.gnu.org/cgit/m4.git/commit/?h=branch-1.4&id=0ffcd0defd006eccc7d514aa4af1151f288cd220

That update should pull in the fixes from `gnulib` making this patch unnecessary.

So, yes, a 1.4.19 _should_ contain the fix that makes this patch unnecessary, at which point it can be removed from this formula file.

Unfortunately, the original bug was reported back in August, 2018 - https://lists.gnu.org/archive/html/bug-m4/2018-08/msg00000.html

Therefore, I do not know when one of the 2 core maintainers will tag and release a new patch release containing the necessary fix.

> I'd suggest either removing the conditional if OS::Linux::Glibc.system_version >= "2.28" 

I can do that, though I don't know what would happen in `linuxbrew-core` if an older glibc is not compatible with this patch.